src/evaluator/datatypes.ts

```diff
@@ -31,3 +31,11 @@ export const make_primitive = (fun: (...args: EvalValue[]) => EvalResult): Primi
   variant: EvalDataType.Primitive,
   fun,
 });
+
+export function instanceOfEvalData(object: any): object is EvalData {
+  // (<any>EvalDataType) is adapted from https://blog.oio.de/2014/02/28/typescript-accessing-enum-values-via-a-string/
+  return Object.keys(EvalDataType).reduce(
+    (prev, curr) => object.variant === (<any>EvalDataType)[curr] || prev,
+    false
+  );
+}
```

Comment:
I wanted to check that the value of the `SBoxed` is of the type `EvalData`. But I had difficulties checking that the value is actually implementing an interface.

In the end I referred to this https://stackoverflow.com/a/14426274/11284745, but I gotta say that the way I am checking it here is also kinda sketchy..

It might also be overkill as I think the only types of `EvalData` we will have are either closures or primitives?